src/taxes/pages/TaxClassesPage/TaxClassesPage.tsx

```diff
@@ -113,112 +122,126 @@ export const TaxClassesPage: React.FC<TaxClassesPageProps> = props => {
-                          <ListItem key={countryRate.id} hover={false}>
-                            <ListItemCell>{countryRate.label}</ListItemCell>
-                            <ListItemCell>
-                              <TaxInput
-                                value={countryRate.value}
-                                change={e =>
-                                  handlers.handleRateChange(
-                                    countryRate.id,
-                                    e.target.value,
-                                  )
-                                }
-                              />
-                            </ListItemCell>
-                          </ListItem>
-                        )) ?? (
-                          <>
-                            <Skeleton />
-                            <VerticalSpacer />
-                          </>
-                        )}
-                      </List>
-                    </>
-                  )}
-                </Card>
-              </div>
+                    ) : (
+                      <>
+                        <CardContent>
+                          <TextField
+                            value={query}
+                            variant="outlined"
+                            onChange={e => setQuery(e.target.value)}
+                            placeholder={intl.formatMessage(
+                              taxesMessages.searchTaxClasses,
+                            )}
+                            fullWidth
+                            InputProps={{
+                              startAdornment: (
+                                <InputAdornment position="start">
+                                  <SearchIcon />
+                                </InputAdornment>
+                              ),
+                            }}
+                            inputProps={{ className: classes.searchPadding }}
+                          />
+                        </CardContent>
+                        <List gridTemplate={["5fr 2fr"]}>
+                          <ListHeader>
+                            <ListItem>
+                              <ListItemCell>
+                                <FormattedMessage
+                                  {...taxesMessages.countryNameHeader}
+                                />
+                              </ListItemCell>
+                              <ListItemCell className={classes.right}>
+                                <FormattedMessage
+                                  {...taxesMessages.taxRateHeader}
+                                />
+                              </ListItemCell>
+                            </ListItem>
+                          </ListHeader>
+                          <Divider />
+                          {filteredRates?.map(
+                            (countryRate, countryRateIndex) => (
+                              <React.Fragment key={countryRate.id}>
```

Comment:
you already have a very big component here, please try to split it into smaller ones. For instance, this map body can be a separate component eg. TaxListItem (find some more meaningful name to your intention) and you would have 

```
...map((...) => <TaxListItem ... />)
```

cleaner

Reply:
I'm going to refactor this during the next taxes update